### ISSUE TITLE
Command injection

### ISSUE THREAD

**Issue by poechsel** [AAA97AAA/SoftSec#6]
The `Date` function call the `date` shell command by using `execlp`:

```C
execlp("date", "date", (char*) NULL);
```

However `execlp` will not always execute a shell command but can also execute a file. From the man:
```
The execlp function duplicate the actions of the shell in searching for an executable file if the specified filename does not contain a slash (/) character.
```

So here, if we `put` an executable file named `date` on the server, a subsequent call to `date` will execute this file and not `/bin/date`.

Luckily for us when we are using `put` the server will call `chmod` on the newly created file with the `0x1ed` mode (executable for everyone).

**Exploit:**

Create a `date` file with the following content:
```
#!/bin/sh
xcalc
```

Execute the following sequence of command:

```
login u1
pass p1
put date 16
date
```

**Comment by hazimeh** [comment]
Good catch; you missed the other half of the puzzle though.
execlp searches the PATH env variable, not the current working dir; chmod alone is not enough to enable this backdoor.
If you look closely, the server changes its PATH env variable elsewhere as well to accommodate this requirement.

**Comment by hazimeh** [comment]
Fixed in commit c322c2e27761e4f7e6fa309e20f1431abbff9862
